sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -169,64 +169,6 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
     }
   }
 
-  /**
-   * Create a table, returning a [[CreateTable]] logical plan.
-   *
-   * This is used to produce CreateTempViewUsing from CREATE TEMPORARY TABLE.
-   *
-   * TODO: Remove this. It is used because CreateTempViewUsing is not a Catalyst plan.
-   * Either move CreateTempViewUsing into catalyst as a parsed logical plan, or remove it because
-   * it is deprecated.
-   */
-  override def visitCreateTable(ctx: CreateTableContext): LogicalPlan = withOrigin(ctx) {
-    val (ident, temp, ifNotExists, external) = visitCreateTableHeader(ctx.createTableHeader)
-
-    if (!temp || ctx.query != null) {
-      super.visitCreateTable(ctx)
```

Comment:
AstBuilder.visitCreateTable should be updated accordingly.  Some comment and code path regarding CREATE TEMPORARY TABLE ... USING ... seems not correct.

```scala
Option(ctx.query).map(plan) match {
  ...
  case None if temp =>
    // CREATE TEMPORARY TABLE ... USING ... is not supported by the catalyst parser.
    // Use CREATE TEMPORARY VIEW ... USING ... instead.
    operationNotAllowed("CREATE TEMPORARY TABLE IF NOT EXISTS", ctx)
  ..
}
```

Reply:
This code path is still correct, right? ```CREATE TEMPORARY TABLE...USING...``` is not supported, only ```CREATE TEMPORARY VIEW...USING...``` is supported.

Reply:
After this PR, `CREATE TEMPORARY TABLE ` is not supported at all. Previously it was supported, though deprecated. Only `CREATE TEMPORARY TABLE IF NOT EXISTS` is not supported at all.

We should update the message `CREATE TEMPORARY TABLE IF NOT EXISTS` in `operationNotAllowed`.

Previously `CREATE TEMPORARY TABLE ... USING ...` is not supported by the catalyst parser but supported by SparkSqlParser. We should update the comment too (remove `by the catalyst parser`).

Reply:
Thanks! Updated.